03_Modules/Certificates/src/module/module.ts

```diff
@@ -312,27 +323,131 @@ export class CertificatesModule extends AbstractModule {
+        await this.sendCall(
+          message.context.stationId,
+          message.context.tenantId,
+          CallAction.GetInstalledCertificateIds,
+          {} as GetInstalledCertificateIdsRequest,
+        );
+      } catch (e: any) {
+        console.error(
+          `There was an error sending GetInstalledCertificateIds message to the charger after installing certificate ${e.message}`,
+          e,
+        );
+      }
+    }
   }
 
   @AsHandler(CallAction.GetInstalledCertificateIds)
-  protected _handleGetInstalledCertificateIds(
+  protected async _handleGetInstalledCertificateIds(
     message: IMessage<GetInstalledCertificateIdsResponse>,
     props?: HandlerProperties,
-  ): void {
+  ): Promise<void> {
     this._logger.debug('GetInstalledCertificateIds received:', message, props);
+    const certificateHashDataList: CertificateHashDataChainType[] =
+      message.payload.certificateHashDataChain!;
+    // persist installed certificate information
+    if (certificateHashDataList && certificateHashDataList.length > 0) {
+      // delete previous hashes for station
+      try {
+        await this._installedCertificateRepository.deleteAllByQuery({
+          where: {
+            stationId: message.context.stationId,
+          },
+        });
+      } catch (error: any) {
+        this._logger.error(
+          'GetInstalledCertificateIds failed to delete previous certificates',
+          error,
+        );
+      }
+      // save new hashes
+      const records = certificateHashDataList.map(
+        (certificateHashDataWrap: CertificateHashDataChainType) => {
+          const certificateHashData =
+            certificateHashDataWrap.certificateHashData;
+          const certificateType = certificateHashDataWrap.certificateType;
+          return {
+            stationId: message.context.stationId,
+            hashAlgorithm: certificateHashData.hashAlgorithm,
+            issuerNameHash: certificateHashData.issuerNameHash,
+            issuerKeyHash: certificateHashData.issuerKeyHash,
+            serialNumber: certificateHashData.serialNumber,
+            certificateType: certificateType,
+          } as InstalledCertificate;
+        },
+      );
+      this._logger.info('Attempting to save', records);
+      const response = await this._installedCertificateRepository.bulkCreate(
+        records,
+        Namespace.InstalledCertificate,
+      );
+      if (response.length === records.length) {
+        this._logger.info(
+          'Successfully updated installed certificate information for station',
+          message.context.stationId,
+        );
+      }
+    } else if (
+      message.payload.status === GetInstalledCertificateStatusEnumType.NotFound
+    ) {
+      // charger has no certs
+      this._logger.info(
+        'No certs found on charger, will delete for station',
+        message.context.stationId,
+      );
+      try {
+        await this._installedCertificateRepository.deleteAllByQuery(
```

Comment:
Here is the same reason as in the previous comment, so we cannot simply delete all installed certificates in db. 

Another idea I have is that we can add a new optional field `certificateType` in this `CallMessage` table. Whenever `GetInstalledCertificateIdsRequest` is called, we stores the correlation id and certificateTypes in this `CallMessage` table. Then when getting response, we can retrieve the CallMessage entity to know what types of certificate are requested. An example is here: https://github.com/citrineos/citrineos-core/blob/f0bda14d4aed4b9f6ec800e4407119ca072a31c3/03_Modules/EVDriver/src/module/api.ts#L289

Reply:
for now adjusting module to not automatically trigger GetInstalledCertificateIds, and when GetInstalledCertificateIds payload comes in from the charger, deleting only installed certificate records with matching certificate types.